### PR TITLE
[new release] atdts, atds, atdpy, atdj, atdgen, atdgen-runtime, atdgen-codec-runtime and atd (2.5.0)

### PR DESCRIPTION
**Pull request by mjambon**
New package in this release: atdts

- Project page: <a href="https://github.com/ahrefs/atd">https://github.com/ahrefs/atd</a>

##### CHANGES:

* atdpy: make `atdpy --version` print the version of atdpy itself
  rather than the version of the `atd` library (ahrefs/atd#270)
* atdpy: fix handling of `nullable` and improve error message on
         `option` types used without optional fields (ahrefs/atd#277)
* Add TypeScript backend called atdts (ahrefs/atd#274)

### CODE REVIEW

**Comment by mseri** [comment]
```

#=== ERROR while compiling atdpy.2.5.0 ========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/ahrefs/atd/releases/download/2.5.0/atdts-2.5.0.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/atdpy.2.5.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p atdpy -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/atdpy-7373-5c57ae.env
# output-file          ~/.opam/log/atdpy-7373-5c57ae.out
### output ###
# File "/home/opam/.opam/4.08/lib/biniou/biniou.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.08/lib/atdgen-runtime/atdgen-runtime.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
#     ocamlopt atdpy/src/lib/.atdpy.objs/native/atdpy__Codegen.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -w -27 -safe-string -g -I atdpy/src/lib/.atdpy.objs/byte -I atdpy/src/lib/.atdpy.objs/native -I /home/opam/.opam/4.08/lib/atd -I /home/opam/.opam/4.08/lib/re -I atdgen/src/.atdgen_emit.objs/byte -I atdgen/src/.atdgen_emit.objs/native -intf-suffix .ml -no-alias-deps -open Atdpy -o atdpy/src/lib/.atdpy.objs/native/atdpy__Codegen.cmx -c -impl atdpy/src/lib/Codegen.ml)
# File "atdpy/src/lib/Codegen.ml", line 134, characters 4-24:
# 134 |     Atd.Unique_name.init
#           ^^^^^^^^^^^^^^^^^^^^
# Error: Unbound module Atd.Unique_name

-     ocamlopt atdpy/src/lib/.atdpy.objs/native/atdpy__Codegen.{cmx,o} (exit 2)
- (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -w -27 -safe-string -g -I atdpy/src/lib/.atdpy.objs/byte -I atdpy/src/lib/.atdpy.objs/native -I /home/opam/.opam/4.08/lib/atd -I /home/opam/.opam/4.08/lib/re -I atdgen/src/.atdgen_emit.objs/byte -I atdgen/src/.atdgen_emit.objs/native -intf-suffix .ml -no-alias-deps -open Atdpy -o atdpy/src/lib/.atdpy.objs/native/atdpy__Codegen.cmx -c -impl atdpy/src/lib/Codegen.ml)
- File "atdpy/src/lib/Codegen.ml", line 134, characters 4-24:
- 134 |     Atd.Unique_name.init
-           ^^^^^^^^^^^^^^^^^^^^
- Error: Unbound module Atd.Unique_name
```

```
#=== ERROR while compiling atdts.2.5.0 ========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/ahrefs/atd/releases/download/2.5.0/atdts-2.5.0.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/atdts.2.5.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p atdts -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/atdts-7374-19ff74.env
# output-file          ~/.opam/log/atdts-7374-19ff74.out
### output ###
# File "/home/opam/.opam/4.08/lib/biniou/biniou.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.08/lib/atdgen-runtime/atdgen-runtime.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
#     ocamlopt atdts/src/lib/.atdts.objs/native/atdts__Codegen.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -w -27 -safe-string -g -I atdts/src/lib/.atdts.objs/byte -I atdts/src/lib/.atdts.objs/native -I /home/opam/.opam/4.08/lib/atd -I /home/opam/.opam/4.08/lib/re -I atdgen/src/.atdgen_emit.objs/byte -I atdgen/src/.atdgen_emit.objs/native -intf-suffix .ml -no-alias-deps -open Atdts -o atdts/src/lib/.atdts.objs/native/atdts__Codegen.cmx -c -impl atdts/src/lib/Codegen.ml)
# File "atdts/src/lib/Codegen.ml", line 146, characters 4-24:
# 146 |     Atd.Unique_name.init
#           ^^^^^^^^^^^^^^^^^^^^
# Error: Unbound module Atd.Unique_name
```

File: packages/atdpy/atdpy.2.5.0/opam [hunk 1]
```diff
@@ -62,7 +62,7 @@ bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
   "dune" {>= "2.8"}
   "ocaml" {>= "4.08"}
-  "atd" {>= "2.3.0"}
+  "atd" {= version}
```
Comment (mseri) [inline packages/atdpy/atdpy.2.5.0/opam:65]: You probably already know this: to achieve this in the `dune-project` file, you can use the `(= :version)` constraint

**Comment by mjambon** [comment]
Let's see if the constraint `"atd" {>= version}` solves it.

File: packages/atdpy/atdpy.2.5.0/opam (hunk 1 above)
Comment (mjambon) [reply packages/atdpy/atdpy.2.5.0/opam:65]: Thanks! I just discovered it after a failed attempt.

**Comment by mseri** [comment]
Looks good!

**Comment by mjambon** [comment]
Thanks!

**Pull request merged**